services/download/src/main/java/app/coronawarn/server/services/download/FederationBatchProcessor.java

```diff
@@ -124,8 +127,7 @@ public void processUnprocessedFederationBatches() {
     String batchTag = batchInfo.getBatchTag();
     logger.info("Processing batch for date {} and batchTag {}", date, batchTag);
     try {
-      BatchDownloadResponse response = federationGatewayClient.getDiagnosisKeys(batchTag, date);
-
+      BatchDownloadResponse response = federationGatewayClient.getDiagnosisKeys(batchTag, date).orElseThrow();
```

Comment:
This throws a NoSuchElementException if no response from EFGS which will kill the download process right? Will there be an error / failure log because of this?

Reply:
Yes, there is logging in the catch-block accordingly